client/electron/go_plugin.ts

```diff
@@ -36,7 +34,85 @@ export async function invokeMethod(
   method: string,
   input: string
 ): Promise<string> {
-  if (!invokeMethodFunc) {
+  console.debug(`[Backend] - calling InvokeMethod "${method}" ...`);
+  const result = await ensureCgo().invokeMethod(method, input);
+  console.debug(`[Backend] - InvokeMethod "${method}" returned`, result);
+  if (result.ErrorJson) {
+    throw Error(result.ErrorJson);
+  }
+  return result.Output;
+}
+
+/**
+ * Represents a callback function for an event.
+ * @param data The event data string passed from the source.
+ * @param param The param string provided during registration.
+ */
+export type CallbackFunction = (data: string, param: string) => void;
```

Comment:
As far as I understand you are decoupling callbacks from events like we discussed. So this would have a single data parameter.

Reply:
Yes.